docs/background/mathematical_model.rst

```diff
@@ -55,4 +59,168 @@ be a bit more complicated:
+we define the following set for every slot :math:`j`:
+
+.. math::
+   :label: same_session_set
+
+   {K}_{j} = \{1\leq j' \leq N\,|\,\text{ if }j\text{ and }j'\text{ are in the same session}\}
+
+We also assume that we have a number of collections of events. Note that these
+collections are non disjoint: any event can be in multiple collections. We refer
+to these collections as "tags": an event can for example be tagged as
+"beginner".
+
+Using this we define the following set for every event :math:`i`
+
+.. math::
+   :label: same_tag_event_set
+
+   T_i = \{1\leq i'\leq M\,|\,\text{ if }i\text{ and }j\text{ do not share a tag}\}
+
+This leads us to the following constraint:
+
+.. math::
+   :label: tag_constraint
+
+    X_{ij}  + X_{i'j'} \leq 1 \text{ for all  }j'\in K_j\text{ for all }1\leq j\leq N\text{ for all }i'\in T_i\text{ for all }1\leq i\leq M
+
+
+Expressions :eq:`all_events_scheduled_constraint`,
+:eq:`all_slots_at_most_1_event_constraint`,
+:eq:`slot_constraint`, :eq:`event_constraint` and :eq:`tag_constraint` define a
+valid schedule and can be used by themselves.
+
+However, it might be desirable to also optimise a given objective function.
+
+Objective functions
++++++++++++++++++++
+
+Optimising to avoid room overflow
+---------------------------------
+
+Demand for events might be known: this will be captured using a vector
+:math:`d\in\mathbb{R}_{\geq 0}^{M}`. Similarly capacity for rooms might be
+known, captured using another vector :math:`c\in\mathbb{R}_{\geq 0}^{N}`. Whilst
+it might not be possible to stick to those constraints strongly (when dealing
+with parallel sessions delegate might not go where they originally intended) we
+can aim to minimise the expected overflow given by the following expression:
+
+.. math::
+   :label: overflow_objective_function
+
+   \sum_{i=1}^{M}\sum_{j=1}^{N}X_{ij}(c_j - d_i)
+
+Using this our optimisation problem to give a desirable schedule is obtained by
+solving the following problem:
+
+Minimise :eq:`overflow_objective_function` subject to :eq:`all_events_scheduled_constraint`,
+:eq:`all_slots_at_most_1_event_constraint`,
+:eq:`slot_constraint`, :eq:`event_constraint` and :eq:`tag_constraint`.
+
+Minimise change from a previous schedule
+----------------------------------------
+
+Once a schedule has been obtained and publicised to all delegates a new
+constraint might arise (modifying :eq:`all_events_scheduled_constraint`,
+:eq:`all_slots_at_most_1_event_constraint`,
+:eq:`slot_constraint`, :eq:`event_constraint` and :eq:`tag_constraint`). At this
+point the original optimisation problem can be solved again leading to a
+potentially completely different schedule. An alternative to this is to use
+distance from an original schedule :math:`X_o` as the objective function. Norms
+on matrix spaces are usually non linear however given the boolean nature of our
+variables the following function can be used to measure the number of changes:
```

Comment:
missing commas: 'however, given the boolean nature of our variables, the following function'